### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by sergeybykov**
**What changed?**
Revamped readme for clarity.

**Why?**
Help people understand what Temporal is and how to start with it.

**How did you test it?**
N/A

**Potential risks**
N/A

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,39 +1,65 @@
-# Temporal  
 [![Build status](https://badge.buildkite.com/fc0e676d7bee1a159916af52ebdb541708d4b9f88b8a980f6b.svg)](https://buildkite.com/temporal/temporal-server)
 [![Coverage Status](https://coveralls.io/repos/github/temporalio/temporal/badge.svg?branch=master)](https://coveralls.io/github/temporalio/temporal?branch=master)
 [![Discourse](https://img.shields.io/static/v1?label=Discourse&message=Get%20Help&color=informational)](https://community.temporal.io)
 
-Visit [docs.temporal.io](https://docs.temporal.io) to learn about Temporal.
+# Temporal  
+
+Temporal is a microservice orchestration platform which enables developers to build scalable applications without sacrificing productivity or reliability.
+Temporal server executes units of application logic, Workflows, in a resilient manner that automatically handles intermittent failures, and retries failed operations.
```
Comment (wxing1292) [inline README.md:8]: `Workflows` -> `workflows`

File: README.md [hunk 2]
```diff
@@ -1,39 +1,65 @@
...
 
-This repo contains the source code of the Temporal server. To implement workflows, activities and worker use [Go SDK](https://github.com/temporalio/sdk-go) or [Java SDK](https://github.com/temporalio/sdk-java).
+The programming model of Temporal exposed via the SDKs (currently for Java and Go) offers developers the simple abstractions of Workflows and Activities that can virtually "resume" execution after a failure from the same line of code.
```
Comment (wxing1292) [inline README.md:10]: > "resume" execution after a failure from the same line of code

for workflow, mostly yes
for activity, not really, activity can have the progress of last heartbeat, but not really able to resume from the same line of code

File: README.md [hunk 3]
```diff
@@ -1,39 +1,65 @@
...
 
-See Maxim's talk at [Data@Scale Conference](https://atscaleconference.com/videos/cadence-microservice-architecture-beyond-requestreply) for an architectural overview of Temporal.
+Temporal is a mature technology, a fork of Uber's Cadence.
+Temporal is being developed by [Temporal Technologies](https://temporal.io/), a startup by the creators of Cadence.
+
+[![IMAGE ALT TEXT](http://img.youtube.com/vi/f-18XztyN6c/0.jpg)](http://www.youtube.com/watch?v=f-18XztyN6c "Temporal")
```
Comment (wxing1292) [inline README.md:15]: why the icon is green?

Comment (wxing1292) [reply README.md:15]: http://img.youtube.com/vi/f-18XztyN6c/0.jpg

File: README.md [hunk 4]
```diff
@@ -1,39 +1,65 @@
...
+$ cd docker
+$ docker-compose up
+```
+
+Refer to [Temporal service docker](docker/README.md) for more advanced options.
 
 ### Run the Samples
 
-Try out the sample recipes for [Go](https://github.com/temporalio/samples-go) or [Java](https://github.com/temporalio/samples-java) to get started.
+Clone or download samples for [Go](https://github.com/temporalio/samples-go) or [Java](https://github.com/temporalio/samples-java) and try to run them against the local Temporal server.
```
Comment (wxing1292) [inline README.md:35]: i guess the previous version looks better?

File: README.md (hunk 3 above)
Comment (sergeybykov) [reply README.md:15]: I suspect that's what youtube generated as a still image.

File: README.md [hunk 5]
```diff
@@ -1,39 +1,65 @@
...
 
 ### Use CLI
 
-Try out [Temporal command-line tool](tools/cli/README.md) to perform various tasks on Temporal
+Use [Temporal's command line tool](https://docs.temporal.io/docs/tctl) `tctl` to interact with the local Temporal server.
+
+```bash
+$ tctl namespace list
+$ tctl workflow list
+```
```
Comment (wxing1292) [inline README.md:45]: curious whether we have tctl published to brew / apt / yum / dnf?

File: README.md (hunk 3 above)
Comment (sergeybykov) [reply README.md:15]: I guess we can put our own image instead.

File: README.md [hunk 6]
```diff
@@ -1,39 +1,65 @@
...
+Clone or download samples for [Go](https://github.com/temporalio/samples-go) or [Java](https://github.com/temporalio/samples-java) and try to run them against the local Temporal server.
 
 ### Use CLI
 
-Try out [Temporal command-line tool](tools/cli/README.md) to perform various tasks on Temporal
+Use [Temporal's command line tool](https://docs.temporal.io/docs/tctl) `tctl` to interact with the local Temporal server.
+
+```bash
+$ tctl namespace list
+$ tctl workflow list
```
Comment (wxing1292) [inline README.md:44]:
```zsh
alias tctl="docker run --rm temporalio/tctl:1.1.0 --address host.docker.internal:7233"
tctl namespace list
tctl workflow list
```

File: README.md (hunk 3 above)
Comment (sergeybykov) [reply README.md]: Added a better one. Not showing up in preview for some reason.

File: README.md (hunk 1 above)
Comment (alexshtin) [reply README.md:8]: No, we decided to cap all our important entities. Workflow, Activity, what else?

File: README.md (hunk 4 above)
Comment (alexshtin) [reply README.md:35]: There is no "recipes" any more. I like new version. Just would remove word "try" and rephrase a little.

File: README.md [hunk 7]
```diff
@@ -1,39 +1,66 @@
...
+$ docker-compose up
+```
+
+Refer to [Temporal service docker](docker/README.md) for more advanced options.
 
 ### Run the Samples
 
-Try out the sample recipes for [Go](https://github.com/temporalio/samples-go) or [Java](https://github.com/temporalio/samples-java) to get started.
+Clone or download samples for [Go](https://github.com/temporalio/samples-go) or [Java](https://github.com/temporalio/samples-java) and try to run them against the local Temporal server.
+We have a number of [HelloWorld type scenarios](https://github.com/temporalio/samples-java#helloworld) available.
```
Comment (alexshtin) [inline README.md:36]: There is "hello world" in go too.

File: README.md (hunk 1 above)
Comment (sergeybykov) [reply README.md:8]: I was told we need to capitalize in our docs core concepts like Workflows and Activities.

File: README.md (hunk 6 above)
Comment (alexshtin) [reply README.md:44]: `alias` sounds overwhelming for me. Especially pining it to exact version here. But I agree, we recommend docker-compose few lines above, and it assumes that users won't build anything themselves (let's say they want to play with Java SDK and don't have `go` at all).

File: README.md (hunk 5 above)
Comment (alexshtin) [reply README.md:45]: That would be a best options.

File: README.md [hunk 8]
```diff
@@ -1,39 +1,66 @@
...
 
-Try out [Temporal command-line tool](tools/cli/README.md) to perform various tasks on Temporal
+Use [Temporal's command line tool](https://docs.temporal.io/docs/tctl) `tctl` to interact with the local Temporal server.
+
+```bash
+$ tctl namespace list
+$ tctl workflow list
+```
 
 ### Use Temporal Web
```
Comment (alexshtin) [inline README.md:47]:
```suggestion
### Use Temporal Web UI
```
?

File: README.md (hunk 7 above)
Comment (sergeybykov) [reply README.md:36]: But only one. Is it worth linking to it from here?

File: README.md (hunk 2 above)
Comment (sergeybykov) [reply README.md]: Updated.

File: README.md (hunk 6 above)
Comment (sergeybykov) [reply README.md:45]: I set version to "latest".

File: README.md (hunk 3 above)
Comment (wxing1292) [reply README.md]: for the pic
could you plz setup git lfs?
```zsh
brew install git-lfs
git lfs install
git lfs track "*.png"
```

Comment (wxing1292) [reply README.md]: FYI
in the PR the binary can be shown by click this
image

File: README.md (hunk 1 above)
Comment (wxing1292) [reply README.md:8]: not sure whether `*Workflows*` is better?

File: README.md (hunk 4 above)
Comment (sergeybykov) [reply README.md]: I rephrased a bit.

File: README.md (hunk 7 above)
Comment (sergeybykov) [reply README.md]: I rephrased a bit.

**Review by wxing1292 — APPROVED** [review]

**Review by samarabbas — APPROVED** [review]

**Review by rylandg — APPROVED** [review]
Not perfect but a huge improvement

**Pull request merged**